controllers/ApplicationController.php

```diff
@@ -288,11 +299,33 @@ public function actionMyDelete()
      */
     public function actionDelete($id)
     {
+        throw new GoneHttpException();
         $this->findModel($id)->delete();
-
         return $this->redirect(['my-application']);
     }
 
+    public function actionRequestDeny()
+    {
+        $user = Applicant::findOne(\Yii::$app->user->getIdentity()->id);
+        if (count($user->applications) > 0) {
+            throw new ForbiddenHttpException();
+        }
+        return $this->render('confirm-deny-application');
+    }
+
+    public function actionDeny()
+    {
+        $user = Applicant::findOne(\Yii::$app->user->getIdentity()->id);
+        if (count($user->applications) > 0) {
+            throw new ForbiddenHttpException();
+        }
+        $user->setAttribute('state', 1);
+        $tmp = $user->updateAll(['state' => 1], ['id' => $user->id]);
```

Comment:
Δεν γίνεται έλεγχος εάν ολοκληρώθηκε με επιτυχία η αλλαγή. 
Επίσης να γίνει προτιμότερα με $user->state = Applicant::... και $user->save() 
@jhaniot

Reply:
@spapad Η λύση του save δεν λειτουργεί γιατί αποτυγχάνει το validation (κατά το save βρίσκει duplicate entries για το vat και το identity).  Αυτό μάλλον οφείλεται επειδή υπάρχουν public properties στον Applicant (https://stackoverflow.com/questions/26827002/yii2-active-record-model-not-saving-data). Οπότε το αφήνουμε με το updateAll.